vkcs/blockstorage/resource_volume_test.go

```diff
@@ -301,3 +342,41 @@ resource "vkcs_blockstorage_volume" "volume_1" {
   volume_type = "{{.VolumeType}}"
 }
 `
+
+const testAccBlockStorageVolumeMetadata = `
+resource "vkcs_blockstorage_volume" "volume_1" {
+  name = "volume_1"
+  description = "metadata test volume"
+  metadata = {
+    key1 = "val0"
+    key2 = "val2"
+	key3 = "val3"
+  }
+  size = 1
+  availability_zone = "{{.AvailabilityZone}}"
+  volume_type = "{{.VolumeType}}"
+}`
+
+const testAccBlockStorageVolumeMetadataUpdate = `
+resource "vkcs_blockstorage_volume" "volume_1" {
+  name = "volume_1"
+  description = "metadata test volume"
+  metadata = {
+	key1 = "val1"
+    key3 = "val3"
+	key4 = "val4"
+  }
+  size = 1
+  availability_zone = "{{.AvailabilityZone}}"
+  volume_type = "{{.VolumeType}}"
+}`
+
+const testAccBlockStorageVolumeMetadataDelete = `
+resource "vkcs_blockstorage_volume" "volume_1" {
+  name = "volume_1"
+  description = "metadata test volume"
+  metadata = {}
```

Comment:
Isn't it better to test unconfigured (null) value? I supposed that a rare user would set metadata to an empty dict instead of removing it if they wanted to unset it.

Reply:
done